src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -3011,6 +3013,27 @@ private function registerFeatureFlagsConfiguration(array $config, ContainerBuild
             ->addTag('feature_flags.feature_strategy')
         ;
 
+        $container->registerAttributeForAutoconfiguration(AsStrategy::class,
+            static function (ChildDefinition $definition, AsStrategy $attribute, \ReflectionClass|\ReflectionMethod $reflector): void {
+                if ($reflector instanceof ReflectionClass) {
+                    if (!$reflector->hasMethod('__invoke')) {
+                        throw new \LogicException('Wrong');
+                    }
+
+                    $callbackMethod = '__invoke';
+                } else {
+                    $callbackMethod = $reflector->getName();
+                }
```

Comment:
Pas sûr de comprendre. Si c'est sur une classe, c'est forcément `__invoke` ?

Je le verrai plus comme ça :
```suggestion
                $callbackMethod = $reflector instanceof ReflectionMethod ? $reflector->getName() : ($attribute->method ?? '__invoke');
                if (!$reflector->hasMethod($callbackMethod)) {
                    throw new \LogicException(sprintf('Invalid strategy method. Method "%s" not found in "%s" class.', $callbackMethod, $reflector->getClassName()));
                }
                // Check method return ?
```

Reply:
dans le else c'est pas une classe c'est une methode. du coup je recup le name de la methode. et je comprend pas l'interet du method sur l'attribute... je trouve ca rend plus confus et surtotu pas tres utile : on peut le mettre aussi bien sur une methode que sur une classe. je vois pas quel probleme ca regle d'avori une property method.

Reply:
Ça peut servir pour des méthodes héritées par exemple. C'est déjà le cas pour le `#[AsMessageHandler]`.

Reply:
hmmm... ok... j'ai du mal avec ca mais en effet... ok je rajoute. Dans le meme esprit que les event listeners je pensais du coup a un FeatureStratSubscriber (name pending) mais qui du coup idem permettrais de configure plusieurs facilement. a voir sur une autre MR si pertinent ou non.

Reply:
done